modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/agg/MapReduceAggregates.java

```diff
@@ -645,8 +645,15 @@ private static MapReduceAgg createAvgAgg(
 
                 sumDivCnt = rexBuilder.makeCall(IgniteSqlOperatorTable.DECIMAL_DIVIDE, numeratorRef, denominatorRef, p, s);
             } else {
-                RexNode divideRef = rexBuilder.makeCall(SqlStdOperatorTable.DIVIDE, numeratorRef, denominatorRef);
-                sumDivCnt = rexBuilder.makeCast(call.getType(), divideRef, true, false);
+                RelDataType resultType = typeFactory.decimalOf(call.type);
```

Comment:
I propose to use single code for both cases without any branches. Also use ``false`` for ``matchNullability`` in cast - it needs only for literals, but it's not literals in the case.